lib/ansible/executor/module_common.py

```diff
@@ -1334,13 +1350,21 @@ def get_action_args_with_defaults(action, args, defaults, templar):
         module_defaults = templar.template(module_defaults)
 
         # deal with configured group defaults first
-        if action in C.config.module_defaults_groups:
-            for group in C.config.module_defaults_groups.get(action, []):
-                tmp_args.update((module_defaults.get('group/{0}'.format(group)) or {}).copy())
+        for group_name in collection_groups:
+            if 'group/%s' % group_name not in module_defaults:
+                continue
+
+            for collection_name in collection_groups[group_name]:
+                collection_routing = _get_collection_metadata(collection_name).get('action_groups', {})
+
+                for candidate_name in redirected_names:
+                    if candidate_name in collection_routing:
```

Comment:
personal preference, but since we only want to do the work once based on the result of a looped query, I usually write those things as an explicit query predicate, eg:
```
if any(name for name in redirected_names if name in action_group):
    tmp_args.update(...)
```
this form also stops the iteration once we know there's work to be done. You could accomplish the same in the existing form by adding a `break` after the update, but IMO the existing form is less clear that you're not really looping over the side effect.